src/moment_kinetics.jl

```diff
@@ -75,24 +75,38 @@ using .time_advance: setup_time_advance!, time_advance!
 main function that contains all of the content of the program
 """
 function run_moment_kinetics(to::TimerOutput, input_dict=Dict())
-    # set up all the structs, etc. needed for a run
-    mk_state = setup_moment_kinetics(input_dict)
-
-    # solve the 1+1D kinetic equation to advance f in time by nstep time steps
-    if run_type == performance_test
-        @timeit to "time_advance" time_advance!(mk_state...)
-    else
-        time_advance!(mk_state...)
-    end
-
-    # clean up i/o and communications
-    # last 2 elements of mk_state are `io` and `cdf`
-    cleanup_moment_kinetics!(mk_state[end-1:end]...)
-
-    if block_rank[] == 0 && run_type == performance_test
-        # Print the timing information if this is a performance test
-        display(to)
-        println()
+    try
+        # set up all the structs, etc. needed for a run
+        mk_state = setup_moment_kinetics(input_dict)
+
+        try
+            # solve the 1+1D kinetic equation to advance f in time by nstep time steps
+            if run_type == performance_test
+                @timeit to "time_advance" time_advance!(mk_state...)
+            else
+                time_advance!(mk_state...)
+            end
+        finally
+
+            # clean up i/o and communications
+            # last 2 elements of mk_state are `io` and `cdf`
+            cleanup_moment_kinetics!(mk_state[end-1:end]...)
+        end
+
+        if block_rank[] == 0 && run_type == performance_test
+            # Print the timing information if this is a performance test
+            display(to)
+            println()
+        end
+    catch e
+        # Stop code from hanging when running on multiple processes if only one of them
+        # throws an error
+        if global_size[] > 1
+            println(e)
```

Comment:
> It would be useful to improve the error messages

Something like this?
```suggestion
            println("Abort called on rank $(block_rank[]) due to error. Error message "
                    * "was:\n ", e)
```